stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/handler/MediaAttachmentPreviewHandler.kt

```diff
@@ -31,21 +31,14 @@ public class MediaAttachmentPreviewHandler(private val context: Context) : Attac
         val mimeType = attachment.mimeType
         val type = attachment.type
 
-        if (assetUrl.isNullOrEmpty()) return false
-
-        // Check if the base of Attachment.mimeType is audio or video
-        if (!mimeType.isNullOrEmpty() && (mimeType.contains(ModelType.attach_audio) || mimeType.contains(ModelType.attach_video))) return true
-
-        // If the previous check fails check Attachment.type for file type
-        if (type.isNullOrEmpty()) return false
-        else if (type == ModelType.attach_audio || type == ModelType.attach_video) return true
-
-        // Fallback in case we receive an incomplete mime type and both previous checks fail
-        return if (mimeType == null) {
-            false
-        } else {
-            val supportedMimeSubTypes = buildMimeSubTypeList()
-            supportedMimeSubTypes.any { subtype -> subtype in mimeType }
+        return when {
+            assetUrl.isNullOrEmpty() -> false
+            mimeType?.contains(ModelType.attach_audio) == true -> true
+            mimeType?.contains(ModelType.attach_video) == true -> true
+            type?.contains(ModelType.attach_audio) == true -> true
+            type?.contains(ModelType.attach_video) == true -> true
+            buildMimeSubTypeList().any { subtype -> mimeType?.contains(subtype) == true } -> true
```

Comment:
Again, less optimized - we build a list that allocates memory and then we have an O(N) check if the type exists. It's much more optimized to just return false if the mimeType is `null`.

Reply:
This condition can be added before building the list of subtypes (As you did on the PR ;) )
The idea with this rule is not "only" optimized the logic, but to reduce the number of different conditions a method can run. Having only one return condition, it is easier to modify it later by adding/removing some of them